src/Services/Manager.vala

```diff
@@ -37,18 +37,28 @@ public class Bluetooth.Services.ObjectManager : Object {
     public bool has_object { get; private set; default = false; }
     public bool retrieve_finished { get; private set; default = false; }
 
-    private bool is_discoverable = false;
+    private bool _discoverable = false;
+    public bool discoverable {
+        get {
+            return _discoverable;
+        }
+        set {
+            lock (adapters) {
+                _discoverable = value;
+                foreach (var adapter in adapters.values) {
+                    adapter.discoverable = value;
+                }
+            }
+        }
+    }
+
```

Comment:
These days, I prefer a simple property declaration `public bool discoverable {get; set;} ` combined with a `notify["discoverable"].connect ()` closure.  This saves the private member `_discoverable` and allows property binding.